src/routes/clouds.js

```diff
@@ -0,0 +1,77 @@
+/*******************************************************************************
+ * Copyright (C) 2018 Orange.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Rombit - initial API and implementation
+ ******************************************************************************/
+const express = require('express');
+const router = express.Router();
+const _ = require('lodash');
+const MongoQS = require('mongo-querystring');
+
+const qs = new MongoQS();
+
+const OwnCloud = require('../clouds/owncloud');
+const { Record } = require('../models');
+
+router.route('/')
+  .get((req, res, next) => {
+    res.send({
+      "clouds":[
+        {
+          "owncloud": {
+            "implemented": true
+          },
+          "dropbox": {
+            "implemented": false
+          },
+          "googleDrive": {
+            "implemented": false
+          }
+        }
+      ]
+    });
+  });
+
+router.route('/:cloudId')
+  .get((req, res, next) => {
+    if (req.params.cloudId === "owncloud"){
+      res.send(OwnCloud.endpointDescription());
+
+    }else if (req.params.cloudId === "dropbox") {
+      res.status(501).send('Dropbox Cloud not implemented');
+    }else if (req.params.cloudId === "googledrive") {
+      res.status(501).send('Google Drive Cloud not implemented');
+    }else {
+      res.status(406).send('Unknown cloud');
+    }
+    res.send("hello world");
+  })
+  .post((req, res, next) => {
+    if (req.query) {
+      const query = qs.parse(req.query);
+      if (req.body.credentials === undefined || req.body.credentials.clientId === undefined || req.body.credentials.clientSecret === undefined || req.body.credentials.owncloudServer === undefined){
```

Comment:
Same comment regarding checking for values being undefined.
Also if `credentials` is undefined, checking for `credentials.clientId`, `credentials.clientSecret`, and `credentials.ownCloudServer` will throw uncaught errors.

Reply:
Fixed.

Reply:
@Exulansis : the `if` is right thanks to short-circuit evaluation.